olp-cpp-sdk-dataservice-read/tests/JsonResultParserTest.cpp

```diff
@@ -0,0 +1,152 @@
+  {
+    SCOPED_TRACE("Verify valid partitions responce");
+    auto str = std::stringstream(partitions_string);
+
+    auto responce = olp::parser::parse_result<dr::PartitionsResponse>(str);
+    ASSERT_TRUE(responce.IsSuccessful());
+    ASSERT_EQ(10u, responce.GetResult().GetPartitions().size());
+  }
+  {
+    SCOPED_TRACE("Verify corrupted with additional symbol");
+    auto str = std::stringstream(partitions_string + "_");
+
+    auto responce = olp::parser::parse_result<dr::PartitionsResponse>(str);
+    ASSERT_FALSE(responce.IsSuccessful());
+    ASSERT_EQ(responce.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Unknown);
+    ASSERT_EQ(responce.GetError().GetMessage(), "Fail parsing response.");
+  }
+  {
+    SCOPED_TRACE("Verify merged responses");
+    auto partitions2 =
+        mockserver::ReadDefaultResponses::GeneratePartitionsResponse(2);
+    auto partitions_string2 = olp::serializer::serialize(partitions2);
+
+    auto str = std::stringstream(partitions_string + partitions_string2);
+
+    auto responce = olp::parser::parse_result<dr::PartitionsResponse>(str);
+    ASSERT_FALSE(responce.IsSuccessful());
+    ASSERT_EQ(responce.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Unknown);
+    ASSERT_EQ(responce.GetError().GetMessage(), "Fail parsing response.");
+  }
+  {
+    SCOPED_TRACE("Verify missed last brace");
+    std::string json_input =
+        partitions_string.substr(0, partitions_string.size() - 1);
+
+    auto str = std::stringstream(json_input);
+    auto responce = olp::parser::parse_result<dr::PartitionsResponse>(str);
+    ASSERT_FALSE(responce.IsSuccessful());
+    ASSERT_EQ(responce.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Unknown);
+    ASSERT_EQ(responce.GetError().GetMessage(), "Fail parsing response.");
+  }
+}
+
+TEST(JsonResultParserTest, ExtendedResponse) {
+  auto partitions =
+      mockserver::ReadDefaultResponses::GeneratePartitionsResponse();
+  auto partitions_string = olp::serializer::serialize(partitions);
+  using ExtendedResponse =
+      dr::ExtendedApiResponse<dr::PartitionsResult, olp::client::ApiError, int>;
+
+  {
+    SCOPED_TRACE("Verify valid extended partitions responce");
+    auto str = std::stringstream(partitions_string);
+
+    auto responce = olp::parser::parse_result<ExtendedResponse>(str, 100);
+    ASSERT_TRUE(responce.IsSuccessful());
+    ASSERT_EQ(10u, responce.GetResult().GetPartitions().size());
+    ASSERT_EQ(100, responce.GetPayload());
+  }
+  {
+    SCOPED_TRACE("Verify corrupted with additional symbol");
+    auto str = std::stringstream(partitions_string + "_");
+
+    auto responce = olp::parser::parse_result<ExtendedResponse>(str, 100);
+    ASSERT_FALSE(responce.IsSuccessful());
+    ASSERT_EQ(responce.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Unknown);
+    ASSERT_EQ(responce.GetError().GetMessage(), "Fail parsing response.");
+  }
+}
+
+TEST(JsonResultParserTest, WithArgs) {
```

Comment:
This is basically the same as ExtendedResponse

Reply:
It not the same, ExtendedResponse has additional arg not as Result, here we have Result that combined from parsed type and other types.